core/server/common/src/main/java/alluxio/master/audit/UserAccessAuditLog.java

```diff
@@ -0,0 +1,59 @@
+package alluxio.master.audit;
+
+import alluxio.RpcUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ArrayBlockingQueue;
+
+public class UserAccessAuditLog {
+  private static final int QUEUE_SIZE = 100;
+  private static final Logger LOG = LoggerFactory.getLogger(UserAccessAuditLog.class);
+  private boolean mEnabled;
+  private ArrayBlockingQueue<RpcUtils.RpcContext> mAuditLogEntries;
+
+  public UserAccessAuditLog() {
+    mEnabled = true;
+    mAuditLogEntries = new ArrayBlockingQueue<>(QUEUE_SIZE);
+  }
+
+  public boolean isEnabled() { return mEnabled; }
+
+  public boolean append(RpcUtils.RpcContext context) {
+    try {
+      mAuditLogEntries.put(context);
+    } catch (InterruptedException e) {
+      // TODO
+      e.printStackTrace();
+    }
+    return true;
+  }
+
+  public void commit(RpcUtils.RpcContext context) {
+    synchronized (context) {
```

Comment:
In the new implementation based on Closeable, we may be able to do without this. Let me double check.